File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = dict()
        for i, num in enumerate(nums):
            if num in lookup:
                lookup[num].append(i)
            else:
                lookup[num] = [i]
        print(lookup)
        for i, num in enumerate(nums):
            diff = target - num
            print(f"diff {diff}")
            if diff in lookup:
                for j in lookup[diff]:
                    if j != i:
                        return [i, j]
        return []

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.rs
impl Solution {
    pub fn product_except_self(nums: Vec<i32>) -> Vec<i32> {
        let mut out = Vec::new();
        for i in 0..nums.len() {
            out.push(1);
            for j in 0..nums.len() {
                if i == j {
                    continue;
                }
               out[i] *= nums[j];
            }
        }
        out
    }
}
